src/compiler/parser.ts

```diff
@@ -3793,12 +3797,12 @@ namespace ts {
             const unaryOperator = token();
             const simpleUnaryExpression = parseSimpleUnaryExpression();
             if (token() === SyntaxKind.AsteriskAsteriskToken) {
-                const start = skipTrivia(sourceText, simpleUnaryExpression.pos);
+                const range = { pos: skipTrivia(sourceText, simpleUnaryExpression.pos), end: simpleUnaryExpression.end };

```

Comment:
why not use `parseErrorAt` here instead of creating a throwaway structure? I don't think readability is that much better.